### PR TITLE
Use models

### PR DESCRIPTION
**Pull request by bmerry**
It's a whole grab-bag of changes:
- Schema update for s3_config.json to specify location of models.
- Put that location into sdp_model_base_url in telstate.
- Follow aliases from that location to determine relative URLs for band mask and RFI models and put those in telstate.
- For simulations, put the band in sub_band in telstate (for the benefit of cal) - previously this was done by katcbfsim, but it was just guessing.
- Switch to async version of TelescopeState.
- During postprocessing, fetch the RFI mask model and use it to skip spectral imaging tasks which are entirely masked, and indicate the processed channels for the report.
- Fix some type annotations and documentation that _get_targets can no longer return None.
- Sandbox script to populate minio with models.

It will fail if the models are not present, so should only be merged once we have some in place.

### CODE REVIEW

File: katsdpcontroller/generator.py [hunk 1]
```diff
@@ -1652,10 +1664,11 @@ def _make_spectral_imager(g: networkx.MultiDiGraph,
         logger.info('Spectral imager targets for %s: %s', capture_block_id,
                     ', '.join(target.name for target, _ in targets))
     view = telstate.view(telstate.join(capture_block_id, stream.name))
-    view['targets'] = {target.description: target_mapper(target) for target, _ in targets}
-    # While it is currently just a contiguous range, output the channels as a
-    # list to allow for applying a static mask in future.
-    view['output_channels'] = list(range(stream.output_channels[0], stream.output_channels[1]))
+    await view.set('targets', {target.description: target_mapper(target) for target, _ in targets})
+    await view.set('output_channels', [
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpcontroller/generator.py:1668]: Not needed. This is just to inform the report writer of which channels it should expect results for.

File: katsdpcontroller/test/test_product_controller.py [hunk 2]
```diff
@@ -230,9 +239,113 @@ def test_already_split(self) -> None:
...
     """Utilities for test classes"""
 
+    def _setup_model(self, model: katsdpmodels.models.Model,
+                     current_path: str,
+                     config_path: str,
+                     fixed_path: str) -> None:
+        fh = io.BytesIO()
+        model.to_file(fh, content_type='application/x-hdf5')
+        self.aioresponses.get(
+            f'http://models.s3.invalid{current_path}',
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpcontroller/test/test_product_controller.py]: Ok, done.

**Comment by bmerry** [comment]
@sratcliffe are you happy for this to be merged now, so that next time we deploy it'll go live? It will need the s3_config.json on site updated when we do.

**Comment by sratcliffe** [comment]
OK - remind me what we need to add to s3_config.json ?

**Comment by bmerry** [comment]
> OK - remind me what we need to add to s3_config.json ?

The models section:
```
    "models": {
        "read": {
            "url": "https://sdpmodels.kat.ac.za/"
        }
    }
```

**Pull request merged**